modules/code-builder/docs/PRD.md

```diff
@@ -0,0 +1,177 @@
+   - **Dedicated Editors/Dialogs**: For more complex properties, a separate interface allows detailed configuration.
+   ![alt text](./images/image-5.png)
+   - **Resizing and Rotation**: Bricks can be resized and rotated to fit the workspace better.
+   - **Cloning/Duplication**: Users can easily create copies of bricks for repeated use.
+   - **Context Menus**: Right-click options for advanced customization and settings.
+
+## 2. Stack of Bricks
+
+### a. Stack Validation:
+   - **Visual Feedback**: Indicators show whether brick combinations are valid.
+
+      ![alt text](./images/image-6.png)
+   - **Error Indicators**: Explanations for incompatible connections help users troubleshoot.
+        - Add a reddish boundary for users to easily tell whether the bricks are mergeable or not.  
+   ![alt text](./images/image-7.png) 
+
+
+   - **Disable Validation**: Temporarily turn off validation for complex or experimental setups.
+
+### b. Stack Editing:
+   - **Connection Editing**: Options for rerouting, splitting, or merging connections.
+   - **Quick Edit Shortcuts**: Context menus or keyboard shortcuts speed up editing.
+   - **Undo/Redo**: Users can revert or reapply changes to their stacks.
+
+### c. Stack Grouping:
+   - **Grouping Bricks**: Multiple bricks or connections can be grouped for better organization.
+   ![alt text](./images/image-8.png)
+   - **Visual Boundaries**: Outlines or boundaries define groups clearly.
+   - **Collapsible Groups**: Groups can be collapsed or expanded to manage complexity.
+
+     ![alt text](./images/image-9.png)
+
+
+## 3. Palette:
+### a. **Layout and Organization**:
+   - Collapsible/expandable categories or sections for different Brick types
+   - Customizable order and arrangement of categories
+   - Visual separators or dividers between categories
+
+      ![alt text](./images/image-10.png)
+
+### b. **Brick Previews**:
+   - Thumbnail or icon previews for each Brick within a category
+   - Tooltips or pop-ups displaying Brick names and brief descriptions
+   - Color-coding or visual cues for distinguishing different Brick types
+
+### c. **Search and Filtering**:
+   - Search bar or input field for locating specific Bricks by name or keyword
+   - Filter options for narrowing down Bricks based on category, type, or properties
+   - Live search results or suggestions as the user types
+
+      ![alt text](./images//image-11.png)
+
+### d. **Drag and Drop**:
+   - Ability to drag and drop Bricks from the palette onto the workspace
+   - Visual indicators (e.g., ghost preview, outline) for valid drop locations
+   - Snap-to-grid or precise positioning when dropping Bricks
+
+        ![alt text](./images/image-12.png)
+
+### e. **Brick Creation**:
+   - Options for creating new Bricks or Brick instances directly from the palette
+   - Context menus or shortcuts for duplicating or cloning existing Bricks
+   - Visual feedback or animations when creating or instantiating new Bricks
```

Comment:
I added this point in regards to a bi-weekly meet where it was discussed to have some kind of custom thing there users can create bricks(ex, Happy birthday some-one wants to play, so user can instantiate that stack of bricks required to create a Happy birthday program in MB and then name it "Happy-birthday-brick" and drag it) from the palette itself. 
This would be a feature that may or may not be implemented but it came across my mind so I wrote it down